src/types.ts

```diff
@@ -0,0 +1,115 @@
+import type { Region } from '@saucelabs/testcomposer';
+
+export type Browser = 'chromium' | 'firefox' | 'webkit' | 'chrome';
+
+export interface Metrics {
+  name: string;
+  data: {
+    install: {duration: number};
+    rebuild?: {duration: number};
+    setup: {duration: number};
+  };
+}
+
+export interface RunResult {
+  startTime: string;
+  hasPassed: boolean;
+  endTime: string;
+  metrics: Metrics[];
+}
+
+export interface RunnerConfig {
+  // NOTE: Kind is serialized by saucectl with a capital 'K' ¯\_(ツ)_/¯
+  Kind: 'playwright';
+  sauce: {
+    region?: Region;
+    metadata?: {
+      build?: string;
+      tags?: string[];
+    };
+  };
+
+  suites: Suite[];
+
+  playwright: {
+    configFile?: string;
+    version: string;
+  };
+
+  // WARN: The following properties are set dynamcially by the runner and are not
+  // deserialized from the runner config json. They should technically be marked
+  // as optional, but the runners treat them as required so type them as such.
+  assetsDir: string;
+  junitFile: string;
+  sauceReportFile: string;
+  preExecTimeout: number;
+  path: string;
+  projectPath: string;
+  playwrightOutputFolder: string;
+  suite: Suite;
+
+  args: Record<string, unknown>;
+}
+
+export interface Suite {
+  name: string;
+  param: SuiteConfig;
+  testMatch: string[] | string;
+  platformName: string;
+  env?: Record<string, string>;
+  preExec: string[];
+  // BUG: The implementation assumes string, but it must be string[]
+  testIgnore?: string;
```

Comment:
🤔 Could you elaborate?
`saucectl`'s config says it's a string array and only accepts those.
I know playwright [accepts a string or string array](https://playwright.dev/docs/api/class-testproject#test-project-test-ignore).

Since this field describes the suite, isn't `string[]` fine?

Reply:
Oops, that was meant to be a note to myself while I was adding the type defs. Yes, it def should be a `string[]`. The current implementation is a bug.